src/phoenix/trace/tracer.py

```diff
@@ -56,6 +56,7 @@ def create_span(
         status_message: Optional[str] = "",
         parent_id: Optional[SpanID] = None,
         trace_id: Optional[UUID] = None,
+        span_id: Optional[SpanID] = None,
```

Comment:
Is the suggestion to use a separate ID for spans than is used for events? I guess we could do that, but we're already exposing trace ID.

Reply:
yea, we should keep the span_id as internal to this function, so there is just one fewer thing to worry about. For trace id, we don't really have a choice because this function operates at the span level

Reply:
This will complicate the implementation in the LlamaIndex case because I'll have to do a tree-traversal rather than a for-loop. I'm not clear on the benefit. Is the purpose to prevent the caller from putting in a non-UUID by mistake?

Reply:
addressed in [0c39152](https://github.com/Arize-ai/phoenix/pull/1058/commits/0c391523f8ad2198d8281e3ce35b456e6960f913)

Reply:
The benefit is that the Tracer class will have fewer "holes". Because the Tracer class is shared by many, it has no obligation to ensure easiness for every implementation, but it does have an obligation to be correct. By plugging up "holes", it's more likely to be correct, even though it comes at the expense of some implementations. It's "selfish" for the Tracer, but that's how it should be.

Reply:
Makes sense